src/components/Board.js

```diff
@@ -3,33 +3,44 @@ import './Board.css';
 import Square from './Square';
 import PropTypes from 'prop-types';
 
+const generateSquareComponents = (squares, updateSquares) => {
+  const singleArray = [];
 
-const generateSquareComponents = (squares, onClickCallback) => {
-  // Complete this for Wave 1
-  // squares is a 2D Array, but 
-  //  you need to return a 1D array
-  //  of square components
+  for (let subArray = 0; subArray < squares.length; subArray++) {
+    squares[subArray].forEach((square) => {
+      singleArray.push(square);
+    });
+  }
 
-}
+  const squaresArray = singleArray.map((square) => {
+    return (
+      <Square
+        key={square.id}
+        id={square.id}
+        value={square.value}
+        updateSquares={updateSquares}
+      ></Square>
+    );
+  });
+  return squaresArray;
+};
+
+const Board = ({ squares, updateSquares }) => {
+  const squareList = generateSquareComponents(squares, updateSquares);
 
-const Board = ({ squares, onClickCallback }) => {
-  const squareList = generateSquareComponents(squares, onClickCallback);
-  console.log(squareList);
-  return <div className="grid" >
-    {squareList}
-  </div>
-}
+  return <div className="grid">{squareList}</div>;
+};
 
 Board.propTypes = {
   squares: PropTypes.arrayOf(
     PropTypes.arrayOf(
       PropTypes.shape({
         id: PropTypes.number.isRequired,
-        value: PropTypes.string.isRequired
+        value: PropTypes.any.isRequired,
```

Comment:
Good job updating prop types! A more specific type you may have considered here is `PropTypes.element.isRequired`, as you're passing an `img` element.